modules/log6x-collection-setup.adoc

```diff
@@ -0,0 +1,201 @@
+
+Setup log collection by binding the required cluster roles to your service account.
+
+== Legacy service accounts
+To use the existing legacy service account `logcollector`, create the following *ClusterRoleBinding*:
+
+[source,terminal]
+----
+$ oc adm policy add-cluster-role-to-user collect-application-logs system:serviceaccount:openshift-logging:logcollector
+$ oc adm policy add-cluster-role-to-user collect-infrastructure-logs system:serviceaccount:openshift-logging:logcollector
+----
+
+Additionally, create the following *ClusterRoleBinding* if collecting audit logs:
+
+[source,terminal]
+----
+$ oc adm policy add-cluster-role-to-user collect-audit-logs system:serviceaccount:openshift-logging:logcollector
+----
+
+
+== Creating service accounts
+.Prerequisites
+
+* The {clo} is installed in the `openshift-logging` namespace.
+* You have administrator permissions.
+
+.Procedure
+
+. Create a service account for the collector. If you want to write logs to storage that requires a token for authentication, you must include a token in the service account.
+
+. Bind the appropriate cluster roles to the service account:
++
+.Example binding command
+[source,terminal]
+----
+$ oc adm policy add-cluster-role-to-user <cluster_role_name> system:serviceaccount:<namespace_name>:<service_account_name>
+----
+
+=== Cluster Role Binding for your Service Account
+The role_binding.yaml file binds the ClusterLogging operator's ClusterRole to a specific ServiceAccount, allowing it to manage Kubernetes resources cluster-wide.
+
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: manager-rolebinding
+roleRef:                                           <1>
+  apiGroup: rbac.authorization.k8s.io              <2>
+  kind: ClusterRole                                <3>
+  name: cluster-logging-operator                   <4>
+subjects:                                          <5>
+  - kind: ServiceAccount                           <6>
+    name: cluster-logging-operator                 <7>
+    namespace: openshift-logging                   <8>
+----
+<1> roleRef: References the ClusterRole to which the binding applies.
+<2> apiGroup: Indicates the RBAC API group, specifying that the ClusterRole is part of Kubernetes' RBAC system.
+<3> kind: Specifies that the referenced role is a ClusterRole, which applies cluster-wide.
+<4> name: The name of the ClusterRole being bound to the ServiceAccount, here cluster-logging-operator.
+<5> subjects: Defines the entities (users or service accounts) that are being granted the permissions from the ClusterRole.
+<6> kind: Specifies that the subject is a ServiceAccount.
+<7> Name: The name of the ServiceAccount being granted the permissions.
+<8> namespace: Indicates the namespace where the ServiceAccount is located.
+
+=== Writing application logs
```

Comment:
This comment applies to the following three sections.

There are prepared ClusterRole resources for "write logs" as well:

- `cluster-logging-write-application-logs`
- `cluster-logging-write-audit-logs`
- `cluster-logging-write-infrastructure-logs`

The user does not need to define their own ClusterRole, but they can do that if they do not want to create ClusterRoleBindings for each of the prepared roles. They need to define their own ClusterRoleBindings to either the prepared roles or a custom role, similar to what they need to do for collection. Those permissions can also be combined into one Role/RoleBinding.

It's possible that it makes more sense that this information is in the "storing logs in LokiStack" part of the documentation.